sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -266,6 +267,10 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
    * @since 1.4.0
    */
   def insertInto(tableName: String): Unit = {
+    extraOptions.get(TimestampTableTimeZone.TIMEZONE_PROPERTY).foreach { tz =>
```

Comment:
we don't seem to be doing this type of validity check in general; otherwise we'd need to add a lot more checks here.

Reply:
The [spec](https://docs.google.com/document/d/1XmyVjr3eOJiNFjVeSnmjIU60Hq-XiZB03pgi3r1razM/edit#heading=h.jnyh4t7bhnlx) requests errors when using invalid time zones. I guess this would still fail with a different error in that case, so I can remove this if you're really against adding it.

Reply:
Hmm. I tried a couple of things, and while it may be possible to remove some of these checks and replace them with a check in `DateTimeUtils.computeTimeZone`, that doesn't cover all cases. For example, you could use "ALTER TABLE" with an invalid time zone and that wouldn't trigger the check.

So given the spec I'm inclined to leave the checks as is, unless @zivanfi is open to making the spec more lax in that area. (`TimeZone.getTimeZone(invalidId)` actually returns the UTC time zone, as unexpected as that behavior may be, so things won't necessarily break without the checks.)

Reply:
Although other table properties don't have similar checks, their effect is usually easy to see. The effect of this specific table property however is not immediately apparent: for new data it is only revealed in interoperability with other components, and for existing data it should not have any visible effect if set correctly. Therefore we decided it would be best to be very strict in checks, because otherwise a typo in the table property value could only be discovered after some data has already been written with irreversible errors. This was the reasoning behind this part of specs.